### PR TITLE
Tweak: Filter out irrelevant templates from ai suggestions [ED-14581]

### PR DESCRIPTION
**Pull request by naorsabag**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Filter out irrelevant templates from ai suggestions 

## Description
An explanation of what is done in this PR

* Filter out irrelevant templates from ai suggestions 

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/ai/module.php [hunk 1]
```diff
@@ -837,11 +837,12 @@ public function ajax_ai_get_layout_prompt_enhancer( $data ) {
 		];
 	}
 
-	private function prepare_generate_layout_context() {
+	private function prepare_generate_layout_context( $data ) {
```
Comment (matipojo) [inline modules/ai/module.php:840]: You changed the signature, did you update all calls to this method?

File: modules/ai/assets/js/editor/ai-layout-behavior.js [hunk 2]
```diff
@@ -24,6 +24,8 @@ export default class AiLayoutBehavior extends Marionette.Behavior {
 	onAiButtonClick( e ) {
 		e.stopPropagation();
 
+		window.elementorAiCurrentContext = this.getOption( 'context' );
```
Comment (matipojo) [inline modules/ai/assets/js/editor/ai-layout-behavior.js:27]: It's overriding the context, maybe you need to restore it after the current action?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/ai/module.php (hunk 1 above)
Comment (naorsabag) [reply modules/ai/module.php:840]: yes

File: modules/ai/assets/js/editor/ai-layout-behavior.js (hunk 2 above)
Comment (naorsabag) [reply modules/ai/assets/js/editor/ai-layout-behavior.js:27]: it gets updated on every AI call, so it will be up to date in the next use

**Review by matipojo — APPROVED** [review]

**Pull request merged**